.github/workflows/linux-compileonly.yaml

```
@@ -35,9 +35,13 @@ jobs:
         shell: bash # for shopt -s dotglob, which also moves hidden files
         run: |
           shopt -s dotglob && mv "$GITHUB_WORKSPACE"/* ${{ github.workspace }}
+      - name: Install CMake
+        run: |
+          wget https://github.com/Kitware/CMake/releases/download/v3.23.0/cmake-3.23.0-linux-x86_64.tar.gz
+          tar -C /usr/local --strip-components=1 -xf cmake-3.23.0-linux-x86_64.tar.gz
```

CI may be failing because we can't extract files into `/usr/local` without root privileges.
Also, fetch the highest version of 3.23.
```suggestion
          wget https://github.com/Kitware/CMake/releases/download/v3.23.5/cmake-3.23.5-linux-x86_64.tar.gz
          sudo tar -C /usr/local --strip-components=1 -xf cmake-3.23.5-linux-x86_64.tar.gz
```